be/src/formats/orc/apache-orc/c++/src/Reader.cc

```diff
@@ -966,8 +978,14 @@ void RowReaderImpl::startNextStripe() {
 
         contents->stream->prepareCache(InputStream::PrepareCacheScope::READ_FULL_STRIPE, currentStripeInfo.offset(),
                                        stripeSize);
+        contents->stream->emptyIORanges();
         currentStripeFooter = getStripeFooter(currentStripeInfo, *contents);
         rowsInCurrentStripe = currentStripeInfo.numberofrows();
+        {
+            std::vector<InputStream::IORange> io_ranges;
+            buildIORanges(&io_ranges);
```

Comment:
If coalesce is disabled, still buildIORanges ?

Reply:
Yes. Almost no performance penalty. But I can make it to build io ranges when only enabled.

Reply:
Fixed.